llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -802,6 +807,8 @@ void PrintIRInstrumentation::printBeforePass(StringRef PassID, Any IR) {
       (shouldPrintBeforePass(PassID) || shouldPrintAfterPass(PassID)))
     DumpIRFilename = fetchDumpFilename(PassID, IR);
 
+  ++CurrentPassNumber;
```

Comment:
this will make pass numbers non-consecutive when we have passes we don't print IR for. is this change necessary? can you find a way to preserve the old behavior? (we should probably add a test that the pass numbers are consecutive)

Reply:
print-at-pass-number.ll does test the pass numbers are consecutive (although it doesn't test passes where the IR isn't printed). If we want to preserve the existing behavior, I can hoist the `shouldPrintIR` call above. The reason for hoisting incrementing `CurrentPassNumber` is so to have the `shouldPrintAfterPass` run after which is needed in the `printAfterPass` call.

Reply:
Moved the `CurrentPassNumber` back to where it was to preserve the behavior of consecutive pass numbers for printing filtered IR. I also added a test for it in the same file. The new solution is to push a new PassRunDescriptor after `CurrentPassNumber` has been incremented.

Reply:
Ugh nevermind, this approach doesn't work as it doesn't clean the PassRunDescriptor stack up. The new approach is to implement a separate check that only pushes the PassRunDescriptor when the you are running with `-print-after-pass-number` option.